src/hotspot/share/classfile/stackMapTable.cpp

```diff
@@ -29,31 +29,25 @@
 #include "oops/oop.inline.hpp"
 #include "runtime/handles.inline.hpp"
 
-StackMapTable::StackMapTable(StackMapReader* reader, StackMapFrame* init_frame,
-                             u2 max_locals, u2 max_stack,
-                             char* code_data, int code_len, TRAPS) {
-  _code_length = code_len;
+StackMapTable::StackMapTable(StackMapReader* reader, TRAPS) {
+  _code_length = reader->code_length();
   _frame_count = reader->get_frame_count();
   if (_frame_count > 0) {
-    _frame_array = NEW_RESOURCE_ARRAY_IN_THREAD(THREAD,
-                                                StackMapFrame*, _frame_count);
-    StackMapFrame* pre_frame = init_frame;
-    for (int32_t i = 0; i < _frame_count; i++) {
-      StackMapFrame* frame = reader->next(
-        pre_frame, i == 0, max_locals, max_stack,
-        CHECK_VERIFY(pre_frame->verifier()));
-      _frame_array[i] = frame;
-      int offset = frame->offset();
-      if (offset >= code_len || code_data[offset] == 0) {
-        frame->verifier()->verify_error(
-            ErrorContext::bad_stackmap(i, frame),
-            "StackMapTable error: bad offset");
-        return;
+    GrowableArray<StackMapFrame*>* tmp_frame_array = new GrowableArray<StackMapFrame*>();
+    while(!reader->at_end()) {
+      StackMapFrame* frame = reader->next(CHECK_VERIFY(reader->pre_frame()->verifier()));
+      if (frame != nullptr) {
+        tmp_frame_array->push(frame);
       }
-      pre_frame = frame;
+    }
+    reader->check_end(CHECK);
+    // Remake frame array with correct size
+    _frame_count = tmp_frame_array->length();
+    _frame_array = NEW_RESOURCE_ARRAY_IN_THREAD(THREAD, StackMapFrame*, _frame_count);
+    for (int i = 0; i < tmp_frame_array->length(); i++) {
+      _frame_array[i] = tmp_frame_array->at(i);
```

Comment:
This leaks the GrowableArray for the rest of the Method verification or until a ResourceMark end scope (where also this StackMapFrame array is also recovered).  Can you just change the type of _frame_array to GrowableArray?  It's the same storage and GrowableArray->length() is the correct count that you can use after this.